File: consolidated_libraries/CommonComponent/src/main/java/com/energy/commoncomponent/bean/RotateDegree.java
package com.energy.commoncomponent.bean;

public enum RotateDegree {

    DEGREE_0(0),

    DEGREE_90(1),

    DEGREE_180(2),

    DEGREE_270(3);

    private final int value;

    RotateDegree(int value) {
        this.value = value;
    }

    public static RotateDegree valueOf(int value) {
        RotateDegree[] types = RotateDegree.values();
        for (RotateDegree type : types) {
            if (type.value == value) {
                return type;
            }
        }
        return DEGREE_0;
    }

    public int getValue() {
        return value;
    }
}

File: libunified/src/main/java/com/mpdc4gsr/libunified/common/RotateDegree.kt
package com.mpdc4gsr.libunified.common
enum class RotateDegree(private val value: Int) {
    DEGREE_0(0),
    DEGREE_90(90),
    DEGREE_180(180),
    DEGREE_270(270);
    fun getValue(): Int {
        return value
    }
}